common/views/components/InfoBanner/InfoBanner.tsx

```diff
@@ -31,6 +34,12 @@ const InfoBanner: FunctionComponent<Props> = ({ cookieName, text }: Props) => {
     setIsVisible(false);
   };
 
+  useEffect(() => {
+    if (prevIsVisible !== isVisible && onVisibilityChange) {
+      onVisibilityChange(isVisible);
```

Comment:
so we have two states , one is more global which we update and can reference too.

Reply:
We have the state of the `InfoBanner` component.

We give access to that property via a callback. The callback is passed from `GlobalInfoBannerContext` which then passes that onto anywhere that wants the value via `useContext`.

![content model(1)](https://user-images.githubusercontent.com/31692/99385777-5b494c80-28c9-11eb-80a7-d9155da2493b.png)